backend/src/resolvers.js

```diff
@@ -12,78 +14,96 @@ export default {
   },
 
   Mutation: {
-    createPost: (parent, args, context) => {
-      // throw error if user does not exist
-      const user = context.dataSources.db.getUser(args.userId);
-      if (!user) {
-        throw new UserInputError('Invalid user', {
-          invalidArgs: 'Userid does not exist',
+    login: (parent, args, context) => {
+      const { email, password } = args;
+
+      if (password.length < 8) {
+        throw new UserInputError('400', {
+          invalidArgs:
+            '[Validation Errors] password is shorter then 8 characters',
         });
       }
 
-      const newPost = {
-        title: args.title,
-        authorid: args.userId,
-        votes: 0,
-      };
+      // get User
+      const user = context.dataSources.db.getUserByEmail(email);
```

Comment:
Email is not validated. It is possible to use an empty string as email address. Possible fix is looking for the '@' symbol or doing a validation check.